File: 351002/Gaiduk/publisher/src/main/java/com/example/publisher/entity/Tweet.java
package com.example.publisher.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "tbl_tweet")
@Data
public class Tweet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, length = 2048)
    private String content;

    @Column(nullable = false)
    private LocalDateTime created;

    @Column(nullable = false)
    private LocalDateTime modified;

    @ManyToMany
    @JoinTable(
            name = "tbl_tweet_sticker",
            schema = "distcomp",
            joinColumns = @JoinColumn(name = "tweet_id"),
            inverseJoinColumns = @JoinColumn(name = "sticker_id")
    )
    private List<Sticker> stickers = new ArrayList<>();
}

File: 351002/Abrashin/src/main/kotlin/com/example/distcomp/model/Tweet.kt
package com.example.distcomp.model

import java.time.LocalDateTime

class Tweet(
    id: Long? = null,
    var creatorId: Long? = null,
    var title: String = "",
    var content: String = "",
    var stickerIds: Set<Long> = emptySet(),
    var created: LocalDateTime? = null,
    var modified: LocalDateTime? = null
) : BaseEntity(id)
